### PR TITLE
Fix FusedConv for ROCm

### PR DESCRIPTION
**Pull request by cloudhan**
1. Fix undesired runtime optimization for non-Relu activation.
2. Refactor
3. Fix false positive runtime error due to fusion failure.

### CODE REVIEW

File: onnxruntime/contrib_ops/rocm/fused_conv.cc [hunk 1]
```diff
@@ -81,41 +82,28 @@ struct FNVHash {
     // miopenConvolutionDescriptor_t, so we have to guess.
     // This algorithm is based on a specific behavior of miopenGetConvolutionNdDescriptor,
     //  which fails when requestedSpatialDim > the convolution's spatial dimension
-    std::vector<int> spatial_dims;
-    std::vector<int> pads;
-    std::vector<int> strides;
-    std::vector<int> dilations;
+    constexpr const int kMaxSpatialDim = 5;
```
Comment (kailums) [inline onnxruntime/contrib_ops/rocm/fused_conv.cc:85]: why limit the Dim to 5? seems original code limit to 10.

File: onnxruntime/contrib_ops/rocm/fused_conv.cc [hunk 2]
```diff
@@ -81,41 +82,28 @@ struct FNVHash {
...
-        break;
-      }
-      spatial_dim += 1;
-      ORT_ENFORCE(spatial_dim < 10,
-                  "miopenGetConvolutionNdDescriptor is supposed to fail before spatial_dim gets to ",
-                  spatial_dim);
+    bool spatial_dim_guessed = false;
+    for (int i=0; i<kMaxSpatialDim; i++) {
+      if (miopenStatusSuccess == miopenGetConvolutionNdDescriptor(
+        cdesc, i, &spatial_dim, pads.data(), strides.data(), dilations.data(), &mode)) {
```
Comment (PeixuanZuo) [inline onnxruntime/contrib_ops/rocm/fused_conv.cc:93]: Why delete vector spatial_dims. I think the meaning of spatial_dim and spatial_dims are different.

Comment (cloudhan) [reply onnxruntime/contrib_ops/rocm/fused_conv.cc:93]: The original one was a miss use of API!

**Comment by kailums** [comment]
LGTM

File: onnxruntime/contrib_ops/rocm/fused_conv.cc [hunk 3]
```diff
@@ -389,17 +371,22 @@ class FusedConv : public onnxruntime::rocm::Conv<T, false> {
 
   static FusionPlanCache plan_cache_;
 
-  Status DoCreateFusionDesc(FusedConvFusionData& fusion) const {
+  Status DoCreateFusionDesc(const std::string node_name, FusedConvFusionData& fusion) const {
```
Comment (abudup) [inline onnxruntime/contrib_ops/rocm/fused_conv.cc:374]: > const std::string node_name

Should this be `const std::string& node_name`? (Note the const-ref).

**Review by abudup — CHANGES_REQUESTED** [review]

**Review by abudup — APPROVED** [review]
LGTM, thank you!

**Pull request merged**